src/Language/Fixpoint/Solver/Interpreter.hs

```diff
@@ -462,7 +465,10 @@ interpret ie γ ctx env   (EApp e1 e2)
   | isSetPred e1                        = let e2' = interpret' ie γ ctx env e2 in 
                                              applySetFolding e1 e2'
 interpret ie γ ctx env e@(EApp _ _)     = case splitEApp e of
-  (f, es) -> let (f':es') = map (interpret' ie γ ctx env) (f:es) in interpretApp ie γ ctx env f' es'
```

Comment:
Maybe
```
-  (f, es) -> let (f':es') = map (interpret' ie γ ctx env) (f:es) in interpretApp ie γ ctx env f' es'
+  (f, es) ->
+    let f' = interpret' ie γ ctx env f
+        es' = map (interpret' ie γ ctx env) es
+     in interpretApp ie γ ctx env f' es'
```
which doesn't need a call to error.

Reply:
I did this but with a local partially applied function.